FirebaseAppCheck/Sources/DeviceCheckProvider/FIRDeviceCheckProvider.m

```diff
@@ -133,10 +134,26 @@ - (void)getTokenWithCompletion:(void (^)(FIRAppCheckToken *_Nullable token,
 #pragma mark - DeviceCheck
 
 - (FBLPromise<NSData *> *)deviceToken {
-  return [FBLPromise
-      wrapObjectOrErrorCompletion:^(FBLPromiseObjectOrErrorCompletion _Nonnull handler) {
-        [self.deviceTokenGenerator generateTokenWithCompletionHandler:handler];
-      }];
+  return [self isDeviceCheckSupported].then(^FBLPromise<NSData *> *(NSNull *ignored) {
+    return [FBLPromise
+        wrapObjectOrErrorCompletion:^(FBLPromiseObjectOrErrorCompletion _Nonnull handler) {
+          [self.deviceTokenGenerator generateTokenWithCompletionHandler:handler];
+        }];
+  });
+}
+
+#pragma mark - Helpers
+
+/// Returns a resolved promise if DeviceCheck is supported and a rejected promise if it is not.
+- (FBLPromise<NSNull *> *)isDeviceCheckSupported {
```

Comment:
How did the SDK behave before this change for this case?

Reply:
If DeviceCheck was unsupported it would produce the following error when calling `limitedUseToken:` or `tokenForcingRefresh:completion:`:
```
Error Domain=com.firebase.appCheck Code=0 "(null)"
UserInfo={
  NSUnderlyingError=0x600000c74000 {
    Error Domain=com.apple.devicecheck.error Code=1 "(null)"
  }
}
```

Reply:
Ahh, good find 👍